builder/delete_test.go

```diff
@@ -1,104 +1,134 @@
+		NamedQuery: "DELETE FROM table USING foobar AS foo",
+	},
+	{
+		Name:       "Using two tables",
+		Builder:    loukoum.Delete("table").Using("foobar", "example"),
+		String:     "DELETE FROM table USING foobar, example",
+		Query:      "DELETE FROM table USING foobar, example",
+		NamedQuery: "DELETE FROM table USING foobar, example",
+	},
+	{
+		Name:       "Using two tables with table statement",
+		Builder:    loukoum.Delete("table").Using(loukoum.Table("foobar"), "example"),
+		String:     "DELETE FROM table USING foobar, example",
+		Query:      "DELETE FROM table USING foobar, example",
+		NamedQuery: "DELETE FROM table USING foobar, example",
+	},
+	{
+		Name:       "Using two tables with as",
+		Builder:    loukoum.Delete("table").Using(loukoum.Table("example"), loukoum.Table("foobar").As("foo")),
+		String:     "DELETE FROM table USING example, foobar AS foo",
+		Query:      "DELETE FROM table USING example, foobar AS foo",
+		NamedQuery: "DELETE FROM table USING example, foobar AS foo",
+	},
+	{
+		Name:       "Where",
+		Builder:    loukoum.Delete("table").Where(loukoum.Condition("id").Equal(1)),
+		String:     "DELETE FROM table WHERE (id = 1)",
+		Query:      "DELETE FROM table WHERE (id = $1)",
+		NamedQuery: "DELETE FROM table WHERE (id = :arg_1)",
+		Args:       []interface{}{1},
+		NamedArgs: map[string]interface{}{
+			"arg_1": 1,
+		},
+	},
+	{
+		Name: "Where time",
+		Builder: loukoum.Delete("table").
 			Where(loukoum.Condition("id").Equal(1)).
-			And(loukoum.Condition("created_at").GreaterThan(when))
-
-		is.Equal("DELETE FROM table WHERE ((id = 1) AND (created_at > '2017-11-23 16:47:27+00'))", query.String())
-	}
+			And(loukoum.Condition("created_at").GreaterThan(now)),
+		String:     fmt.Sprintf("DELETE FROM table WHERE ((id = 1) AND (created_at > %s))", format.Time(now)),
+		Query:      "DELETE FROM table WHERE ((id = $1) AND (created_at > $2))",
+		NamedQuery: "DELETE FROM table WHERE ((id = :arg_1) AND (created_at > :arg_2))",
+		Args:       []interface{}{1, now},
+		NamedArgs: map[string]interface{}{
+			"arg_1": 1,
+			"arg_2": now,
+		},
+	},
+	{
+		Name:       "Returning *",
+		Builder:    loukoum.Delete("table").Returning("*"),
+		String:     "DELETE FROM table RETURNING *",
+		Query:      "DELETE FROM table RETURNING *",
+		NamedQuery: "DELETE FROM table RETURNING *",
+	},
+	{
+		Name:       "Returning id",
+		Builder:    loukoum.Delete("table").Returning("id"),
+		String:     "DELETE FROM table RETURNING id",
+		Query:      "DELETE FROM table RETURNING id",
+		NamedQuery: "DELETE FROM table RETURNING id",
+	},
 }
 
-func TestDelete_Returning(t *testing.T) {
-	is := require.New(t)
-
-	{
-		query := loukoum.Delete("table").Returning("*")
-		is.Equal("DELETE FROM table RETURNING *", query.String())
-	}
-	{
-		query := loukoum.Delete("table").Returning("id")
-		is.Equal("DELETE FROM table RETURNING id", query.String())
+func TestDelete(t *testing.T) {
+	for _, tt := range deletetests {
```

Comment:
Why not define a `type BuilderTestCase func() []BuilderTest` that create a list of BuilderTest so we can do something like that:

```go
func TestDelete(t *testing.T) {
    tests := []BuilderTest{}
    tests = append(tests, CreateDeleteSimpleCase()...)
    tests = append(tests, CreateDeleteUsingCase()...)
    for _, tt := range tests {
        t.Run(tt.Name, tt.Run)
    }
}
```

This avoid the initialization of a huge array, while keeping a great flexibility.

Reply:
The array approach is the recommended way to do it https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test

Reply:
While I agree with that on a small dataset (we use it internally). It's freaking huge having 1200 lines of table driven test.

For the clarity, maintenance and the readability of these unit tests, I think it's best if we split some of those definitions in function...

Reply:
@novln alright, well if we're going to add more functions, then those functions might as well BE the test functions. Why don't we keep the previous function names and grouping, and introduce a:

``` go
type BuilderTests []BuilderTest

func (b BuilderTests) Run(t *testing.T) {
	for _, tt := range b {
		tt.Run(t)
	}
}
```

Then the tests functions would look like this:

``` go

func DeleteSimpleTest(t *testing.T) {
	BuilderTests{
		{
			Name:       "Simple",
			Builder:    loukoum.Delete("table"),
			String:     "DELETE FROM table",
			Query:      "DELETE FROM table",
			NamedQuery: "DELETE FROM table",
		},
		{
			Name:       "Only",
			Builder:    loukoum.Delete("table").Only(),
			String:     "DELETE FROM ONLY table",
			Query:      "DELETE FROM ONLY table",
			NamedQuery: "DELETE FROM ONLY table",
		},
		{
			Name:       "Only",
			Builder:    loukoum.Delete(loukoum.Table("table")).Only(),
			String:     "DELETE FROM ONLY table",
			Query:      "DELETE FROM ONLY table",
			NamedQuery: "DELETE FROM ONLY table",
		},
	}.Run(t)
}
```

Reply:
Yes much better, I prefer the second version as well.